pkg/service/report/report.go

```diff
@@ -204,15 +214,15 @@ func (r *Report) printSummary(reports map[string]*models.TestReport) error {
 			if len(failedList) == 0 {
 				continue
 			}
-			fmt.Printf("\t%s\n", rrow.name)
+			fmt.Fprintf(r.out, "\t%s\n", rrow.name)
 			for _, fc := range failedList {
-				fmt.Printf("\t  - %s\n", fc)
+				fmt.Fprintf(r.out, "\t  - %s\n", fc)
 			}
 		}
 	}
 
-	fmt.Println("<=========================================>")
-	return nil
+	fmt.Fprintln(r.out, "<=========================================>")
+	return r.out.Flush()
```

Comment:
I think you should check the err first then return err by creating fmt.Errof("Failed to print the summary", err) , for better tracing. Do it in all the above returns like this.